packages/core/src/renderer/components/cluster-settings/cluster-settings-menu-injection-token.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { ClusterPreferences } from "../../../common/cluster-types";
+
+export type ChangedClusterPreference = [keyof ClusterPreferences, any];
+
+export interface ClusterIconMenuItem {
+  id: string;
+  title: string;
+  disabled?: (preferences: ClusterPreferences) => boolean;
+  onClick: (preferences: ClusterPreferences) => ChangedClusterPreference;
```

Comment:
Why do we need to return anything on click? Whos gonna use this data?

Reply:
By design, any menu item injectable can do something with preferences object and return changed ones. As we did for `Clear` menu item. It returns part of the preferences which is saved in `icon-settings.tsx`.